core/src/test/scala/org/apache/spark/util/VersionUtilsSuite.scala

```diff
@@ -73,4 +73,29 @@ class VersionUtilsSuite extends SparkFunSuite {
       }
     }
   }
+
+  test("Return short version number") {
+    assert(shortVersion("3.0.0") === "3.0.0")
+    assert(shortVersion("3.0.0-SNAPSHOT") === "3.0.0")
```

Comment:
I didn't change the version example in order to minimize the diff between `master` and `branch-2.4`.